index.bs

```diff
@@ -175,6 +175,173 @@ partial interface Navigator {
+1. Let |audioSession| be |element|'s {{AudioSession}}.
+1. Run |element|'s [=element update steps|update steps=].
+1. If |element| is an [=audible element=] and |audioSession|.[=AudioSession/[[state]]=] is {{AudioSessionState/interrupted}}, run the following steps:
+    1. Add |element| to |audioSession|.[=AudioSession/[[interruptedElements]]=].
+    1. Run |element|'s [=element suspend steps|suspend steps=].
+1. If |element| is in |audioSession|.[=AudioSession/[[interruptedElements]]=], and |audioSession|.[=AudioSession/[[state]]=] is {{AudioSessionState/active}}, run the following steps:
+    1. Remove |element| from |audioSession|.[=AudioSession/[[interruptedElements]]=].
+    1. Run |element|'s [=element resume steps|resume steps=].
+
+When the [=audible flag=] of one of |audioSession|'s [=audio session/elements=] named |changingElement| is changing, the user agent MUST run the following steps:
+1. If |changingElement|'s [=audible flag=] is changing to `true`, [=try activating=] |audioSession| and abort these steps.
+1. Let |shouldAbort| be true if any |element| of |audioSession|.[=AudioSession/[[elements]]=] other than |changingElement| matches any of the following cases:
+    1. |element|'s [=element state|state=] is {{AudioSessionState/interrupted}}.
+    1. |element|'s [=audible flag=] is `true`.
+1. If |shouldAbort| is `true`, abort these steps.
+1. [=Inactivate=] |audioSession|.
+
+## AudioContext ## {#audiocontext-sink}
+
+An {{AudioContext}} is an [=audio session/element=] with the following properties:
+* Its [=default type=] is {{AudioSessionType/ambient}}.
+* Its [=audible flag=] is `true` if its state is {{AudioContextState/running}} and is sending non zero samples to its destination.
+* Its [=element suspend steps|suspend steps=] are:
+    1. Let |audioContext| be the {{AudioContext}} object.
+    1. Queue a control message to suspend |audioContext|.
+* Its [=element resume steps|resume steps=] are:
+    1. Let |audioContext| be the {{AudioContext}} object.
+    1. Queue a control message to unsuspend |audioContext|.
+
+When an {{AudioContext}} is created, the user agent MUST run the following steps:
+1. Let |audioContext| be the newly created {{AudioContext}}.
+1. Let |audioSession| be the {{AudioSession}}'s object of the {{Window}} object in which is created |audioContext|.
+1. Add |audioContext| to |audioSession|.[=AudioSession/[[elements]]=].
+
+## HTMLMediaElement ## {#media-element-sink}
+
+A {{HTMLMediaElement}} is an [=audio session/element=] with the following properties:
+* Its [=default type=] is {{AudioSessionType/playback}}.
+* Its [=audible flag=] is `true` if it is playing, its volume is not <code>0</code>, it is not muted and it has audio tracks.
+* Its [=element suspend steps|suspend steps=] are:
+    1. Let |mediaElement| be the {{HTMLMediaElement}} object.
+    1. [=Queue a task=] to run the internal play steps of |mediaElement|.
```

Comment:
This seems reverse? We would like to want to run `pause` steps for `suspend` and to run `play` steps for `resume`? In addition, could we link those `internal XXX steps` to their HTML definitions?

Reply:
Right, fixed.